File: 53-maximum-subarray/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int max_sum = nums[0];
        for(int i: nums){
            sum+=i;
            if(max_sum<sum) max_sum=sum;
            if(sum<0) sum=0;
        }
        return max_sum;
    }
}

File: 53-maximum-subarray/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans,m = -100000,0
        for i in range(len(nums)):
            m = m+nums[i]
            if(ans<m):
                ans = m
            if(m<0):
                m=0
        return ans
        